modules/surface_matching/src/ppf_helpers.cpp

```diff
@@ -442,6 +442,12 @@ Mat samplePCByQuantization(Mat pc, Vec2f& xrange, Vec2f& yrange, Vec2f& zrange,
         pcData[4]=(float)(ny/norm);
         pcData[5]=(float)(nz/norm);
       }
+      else
+      {
+        pcData[3]=(float)nx;
+        pcData[4]=(float)ny;
+        pcData[5]=(float)nz;
```

Comment:
This may break assumed invariants, like `norm(vec) == 1` for non-zero vectors.
Perhaps we should write here zero values (0,0,0) instead.

/cc @bmagyar

Reply:
You mean set =0 for all case where norm < EPS?

Reply:
Yes

Reply:
Done.